nimble/controller/src/ble_ll_adv.c

```diff
@@ -3405,11 +3405,12 @@ ble_ll_adv_send_conn_comp_ev(struct ble_ll_conn_sm *connsm,
 uint8_t *
 ble_ll_adv_get_local_rpa(struct ble_ll_adv_sm *advsm)
 {
-    uint8_t *rpa;
+    uint8_t *rpa = NULL;
 
-    rpa = NULL;
     if (advsm->own_addr_type > BLE_HCI_ADV_OWN_ADDR_RANDOM) {
-        rpa = advsm->adva;
+        if (ble_ll_is_rpa(advsm->adva, (advsm->own_addr_type & 1))) {
```

Comment:
`advsm->own_addr_type & 1` is probably incorrect since it will always return `false` if `own_addr_type` is `0x02` (RPA or pub)

Reply:
don't forget about 0x03 which is rpa random

Reply:
ah no you are right